cloudinit/config/cc_ssh_import_id.py

```diff
@@ -50,7 +51,12 @@
 
 def handle(_name, cfg, cloud, log, args):
 
-    if is_key_in_nested_dict(cfg, "ssh_import_id") and not subp.which(SSH_IMPORT_ID_BINARY):
+    if any([
+        cfg.get(SSH_IMPORT_ID_CFG_KEY),
+        is_key_in_nested_dict(cfg.get("users"), SSH_IMPORT_ID_CFG_KEY),
+        is_key_in_nested_dict(cfg.get("user"), SSH_IMPORT_ID_CFG_KEY),
```

Comment:
You'll need  either a is_key_in_nested_dict(cfg.get("users", {}) or adapt is_key_in_nested_dict to return False empty or non-iterable values.

```suggestion
        is_key_in_nested_dict(cfg.get("users", {}), SSH_IMPORT_ID_CFG_KEY),
        is_key_in_nested_dict(cfg.get("user", {}), SSH_IMPORT_ID_CFG_KEY),
```

Reply:
I agree, but it isn't quite that simple to solve the failing test. By indexing to users, the return value is a list, which the function isn't handling at the moment. Since this optimization seems premature and others didn't feel strongly about it, I think I'll just revert rather than rewriting `is_key_in_nested_dic()`